src/System.Private.Uri/tests/UriBuilderTests.cs

```diff
@@ -9,6 +9,43 @@ namespace System.Tests
 {
     public class UriBuilderTests
     {
+        //This test tests a case where the Core implementation of UriBuilder differs from Desktop Framework UriBuilder.
+        //The Query property will not longer prepend a ? character if the string being assigned is already prepended.
+        [Fact]
+        public static void TestQuery()
+        {
+            UriBuilder uriBuilder = new UriBuilder(@"http://foo/bar/baz?date=today");
+
+            string q;
+
+            q = uriBuilder.Query;
+            Assert.Equal<String>(q, @"?date=today");
+
+            uriBuilder.Query = @"?date=yesterday";
+            q = uriBuilder.Query;
+            Assert.Equal<String>(q, @"?date=yesterday");
+
+            uriBuilder.Query = @"date=tomorrow";
+            q = uriBuilder.Query;
+            Assert.Equal<String>(q, @"?date=tomorrow");
+
+            uriBuilder.Query += @"&place=redmond";
+            q = uriBuilder.Query;
+            Assert.Equal<String>(q, @"?date=tomorrow&place=redmond");
+
+            uriBuilder.Query = null;
+            q = uriBuilder.Query;
+            Assert.Equal<String>(q, @"");
+
+            uriBuilder.Query = "";
+            q = uriBuilder.Query;
+            Assert.Equal<String>(q, @"");
+
+            uriBuilder.Query = "?";
+            q = uriBuilder.Query;
+            Assert.Equal<String>(q, @"?");
```

Comment:
Assert.Equal's first argument is "expected", seoond is "actual", so all of these Equal calls should have their arguments switched.  It's also not necessary to store Query into a temporary.  And it's not necessary to specify the generic type to Equal, though if it is specified, it should use "string" instead of "String".  So, for example:
```C#
uriBuilder.Query = "?";
q = uriBuilder.Query;
Assert.Equal<string>(q, @"?");
```
becomes:
```C#
uriBuilder.Query = "?";
Assert.Equal("?", uriBuilder.Query);
```